x/ecocredit/server/server.go

```diff
@@ -25,14 +25,31 @@ type serverImpl struct {
 	storeKey sdk.StoreKey
 
 	// we use a single sequence to avoid having the same string/ID identifying a class and batch denom
-	idSeq orm.Sequence
-
+	idSeq          orm.Sequence
 	classInfoTable orm.NaturalKeyTable
-
 	batchInfoTable orm.NaturalKeyTable
 }
 
-func newServer(storeKey sdk.StoreKey) serverImpl {
+// Server is the the ecocredits implementation of ADR 031 Msg Service
+type Server interface {
```

Comment:
Not sure this is desirable. I was trying to not expose this outside of RegisterServices. If we need an interface just inherit MsgServer and QueryServer rather than writing out each method.

Reply:
It helps with testing. The pattern I'm following is to return interfaces wherever possible and following the excellent proverb: 

_“Accept interfaces, return structs”_

There are many benefits from it, especially if we design different components and will like to do unit-tests. It's also part of SOLID principles.

Reply:
But you are right, we should reuse `MsgServer` and `QueryServer`

Reply:
Right I was just wanting to make the serverimpl private and force people to use clients

Reply:
The tests all use clients so they don't need this public

Reply:
Just for context I actually did exactly what you're doing first btw 😉 but then changed to make it private

Reply:
Thanks for the explanation. BTW: interface, not an implementation. But probably you are right - with the comments above, we won't need to expose anything about the server (neither the combined interface). I will look at it tomorrow again, meanwhile I added CLI code (query is done, Tx only one command).